packages/footnotes.lua

```diff
@@ -69,6 +69,11 @@ local function registerCommands (class)
       SILE.settings:set(v, SILE.settings.defaults[v])
     end
 
+    -- Cross-reference suppport.
+    local hasRefs = type(class.pushLabelRef) == "function"
+    if hasRefs then
+      class:pushLabelRef(SILE.formatCounter(SILE.scratch.counters.footnote))
```

Comment:
Lets try to think of a way to let the labelrefs package set a hook that makes footnotes respond this way without having to edit the footnotes package.

Reply:
I don't get it. The laberefs package here provides hooks (push/pop) so that any 3rd party packages that wishes to have labeled items can register them for a duration. Do you mean it should be the other way round? I am not sure I understand how it would work. I am not even using the standard footnotes package (and my footnotes are not even necessarily numbered) - and I have other packages using such hooks (e.g. poetry). Surely the labelrefs package cannot know all of these packages, shouldn't they be the ones responsible for registering their labeled items?

Reply:
Forget what I wrote above. You could certainly introduce a new paradigm and have some class hooks that packages would register to.

Reply:
So, how do we move forward here?

This worked for 0.12.x. I have no idea how to deal with the "requested changes" status.